datasource/read.go

```diff
@@ -37,4 +53,14 @@ type ReadResponse struct {
 	// source. An empty slice indicates a successful operation with no
 	// warnings or errors generated.
 	Diagnostics diag.Diagnostics
+
+	// Deferred indicates that Terraform should defer
+	// reading this data source.
```

Comment:
Nit: I'm not sure if we may want to clarify this against the existing deferred behavior that Terraform will automatically do if you do have a data source that references an unknown value, e.g.

```terraform
data "examplecloud_thing" "example" {
  example = # some unknown value reference
}
```

Which causes the Terraform plan to show a read on apply action for the data source. Maybe one way to do this is to talk about deferral until the next Terraform apply operation?

```suggestion
	// Deferred indicates that Terraform should defer reading this
	// data source until a followup apply operation.
```